packages/edit-site/src/components/page-templates/index.js

```diff
@@ -315,6 +329,50 @@ export default function PageTemplates() {
 				elements: authors,
 				width: '1%',
 			},
+			{
+				header: __( 'Post types' ),
+				id: 'postTypes',
+				getValue: ( { item } ) =>
+					// This logic would be used by the filter
+					// and replicates what we have in the server at
+					// https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/block-template-utils.php#L1077
+					//
+					// 1. Consider the post types defined by the item, if any
+					item.post_types ||
+					// 2. Otherwise, if a template is custom add it for any CPT.
+					( item.is_custom ? [ 'post', 'page' ] : item.post_types ) || // TODO: take any CPT.
```

Comment:
A downside of this approach is that you have to provide all the values that _could_ match. For example, in this case, we should provide all possible CPTs in this array.